examples/CMakeLists.txt

```diff
@@ -12,8 +12,9 @@ macro(add_example EXAMPLE_NAME EXAMPLE_SOURCE)
   add_executable(${EXAMPLE_NAME} ${STATIC_OBJECTS} ${EXAMPLE_SOURCE} ${ARGN})
   target_link_libraries(${EXAMPLE_NAME} open62541 ${open62541_LIBRARIES})
   assign_source_group(${EXAMPLE_SOURCE})
-  add_dependencies(${EXAMPLE_NAME} open62541_amalgamation_header open62541_amalgamation_source)
+  add_dependencies(${EXAMPLE_NAME} open62541_amalgamation)
```

Comment:
This should stay `add_dependencies(${EXAMPLE_NAME} open62541_amalgamation_header open62541_amalgamation_source)`
I had to separate the target to avoid that `open62541.c` is created on every build even if I disable amalgamation. Additionally it solves some dependency issues for CI
Otherwise we can also add an intermediate target `open62541_amalgamation` which includes the `_header` and `_source`